tests/js/click/click-basic-tests.js

```diff
@@ -0,0 +1,143 @@
+
+Licensed under the BSD 3-Clause License. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the BSD 3-Clause License at
+https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
+*/
+
+/* global gamepad, jqUnit */
+
+(function (fluid, $) {
+    "use strict";
+
+    $(document).ready(function () {
+
+        fluid.registerNamespace("gamepad");
+        fluid.registerNamespace("gamepad.tests");
+
+        jqUnit.module("Gamepad Navigator Basic Click Tests", {
+            setup: function () {
+                gamepad.tests.windowObject = window;
+                gamepad.tests.count = 2;
+
+                // To test the gamepad's initial state.
+                gamepad.tests.modelAtRest = {
+                    connected: true,
+                    axes: {},
+                    buttons: {}
+                };
+
+                // Initialize in accordance with the 18 buttons on the PS4 controller.
+                for (var buttonNumber = 0; buttonNumber < 18; buttonNumber++) {
+                    gamepad.tests.modelAtRest.buttons[buttonNumber] = 0;
+                }
+
+                // Initialize in accordance with the 4 axes on the PS4 controller.
+                for (var axesNumber = 0; axesNumber < 4; axesNumber++) {
+                    gamepad.tests.modelAtRest.axes[axesNumber] = 0;
+                }
+
+                jqUnit.expect(5);
+            },
+            teardown: function () {
+                // Destroy the component and verify it.
+                gamepad.tests.navigator.destroy();
+                jqUnit.assertTrue("The instance of the gamepad navigator should be destroyed.", fluid.isDestroyed(gamepad.tests.navigator));
+            }
+        });
+
+        jqUnit.test("Buttons can be clicked.", function () {
+            // Set the initial conditions and requirements.
+            gamepad.tests.clicked = false;
+            gamepad.tests.windowObject.navigator.getGamepads = function () {
+                return gamepad.tests.utils.element.click(0);
+            };
+
+            // Initialize the webpage, i.e., focus on the button element.
+            $("button").focus();
+
+            // Confirm that the instance of the gamepad navigator is created.
+            gamepad.tests.navigator = gamepad.inputMapper({
+                windowObject: gamepad.tests.windowObject
+            });
+            jqUnit.assertTrue("The Gamepad Navigator should be instantiated.", fluid.isComponent(gamepad.tests.navigator));
+
+            // Check the state of gamepad inputs and webpage after polling.
+            gamepad.tests.navigator.pollGamepads();
+            jqUnit.assertLeftHand("The gamepad should be connected with no buttons/axes disturbed initially.", gamepad.tests.modelAtRest, gamepad.tests.navigator.model);
+            jqUnit.assertEquals("The focus should not have changed.", document.querySelector("button"), document.activeElement);
```

Comment:
I don't think you need to duplicate this, it's necessary for and already covered by the focus (tabbing) tests.

Reply:
I had kept it as a verifying point because the tests can also fail due to the fact that the radio, checkbox, button, or any other element isn't focused before we perform the action. But if you still think that we should remove it then please let me know.

Reply:
I see your point, although I dislike that it's literally copied and pasted so many times.  If you can make a helper function for this initial check, that would be better.

Reply:
That's a good suggestion and I think we'd have to do the same for other tests we wrote earlier. I'll address this for other tests in a separate pull, just to keep things simple for you to review.